### PR TITLE
[Enhancement] support trace verbose optimizer and add limit to semi join reorder

### PR DESCRIPTION
**Pull request by packy92**
Signed-off-by: packy <wangchao@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR mainly slove two problems:
1. When there is lots of semi join, the old code still use tranform rule to enumerate plan which may time consuming.  This pr limit the semi/anti join node it the tree. The most base case using tranform rule is a left-deep tree which contians  `CBO_MAX_REORDER_NODE_USE_EXHAUSTIVE inner/cross join node` and  `CBO_MAX_REORDER_NODE_USE_EXHAUSTIVE` semi/anti join node.

2. add `trace verbose optimizer` to record OptimizeTask cost. It records the Execution times and time-consuming for `OptimizeTask`.
image

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 27 / 49 (55.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java|2|21|09.52%|[67, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 97, 98, 100]|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerTraceInfo.java|3|6|50.00%|[37, 70, 71]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/JoinOperator.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/StatementBase.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/StatementBase.java [hunk 1]
```diff
@@ -48,7 +48,8 @@ public enum ExplainLevel {
         VERBOSE,
         // True if the describe_stmt print costs information, if `isCosts` is true, `isExplain` must be set to true.
         COST,
-        OPTIMIZER
+        OPTIMIZER,
+        VERBOSE_OPTIMIZER
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/StatementBase.java:52]: use optimizer directly~ don't need add a verbose

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java [hunk 2]
```diff
@@ -62,12 +63,39 @@ public void executeTasks(TaskContext context) {
...
+        } else if (task instanceof ExploreGroupTask) {
+            timerName = "Optimizer.CostBaseOptimize.ExploreGroupTask";
+        } else if (task instanceof OptimizeExpressionTask) {
+            timerName = "Optimizer.CostBaseOptimize.OptimizeExpressionTask";
+        } else if (task instanceof OptimizeGroupTask) {
+            timerName = "Optimizer.CostBaseOptimize.OptimizeGroupTask";
+        } else if (task instanceof RewriteTreeTask) {
+            timerName = "Optimizer.RuleBaseOptimize.RewriteTreeTask";
+        }
+        try (PlannerProfile.ScopedTimer ignore = PlannerProfile.getScopedTimer(timerName)) {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java:97]: "Optimizer.CostBaseOptimize." + task.getClass().getSimpleName()

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/StatementBase.java (hunk 1 above)
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/StatementBase.java]: > use optimizer directly~ don't need add a verbose

Yes enhance the trace optimizer is better.

Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/ast/StatementBase.java]: > the performance of VERBOSE_OPTIMIZER mode is worse

`trace optimizer` is more like a debug tool. I think the info provided by it is more important than its performance.

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java (hunk 2 above)
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java:87]: Good advice. We need the prefix to represen the hierarchy.

**Comment by packy92** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 36 (69.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java|2|10|20.00%|[67, 81, 82, 83, 85, 87, 88, 90]|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerTraceInfo.java|3|6|50.00%|[37, 70, 71]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Utils.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/JoinOperator.java|2|2|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java (hunk 2 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java:87]: the hierarchy don't depend on the prefix

Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java:87]: Oh，you are right. The level is used for controlling Indentation. But the comment `It's worth to note that, "." in `ComponentA.CompoenentB.FunctionC` is preserved for representing hierarchy` is a little confused.

**Pull request merged**